Anlab.Mvc/Controllers/AdminController.cs

```diff
@@ -2,24 +2,158 @@
         {
             _dbContext = dbContext;
+            _userManager = userManager;
+            _roleManager = roleManager;
         }
-        public IActionResult Index()
+        public async Task<IActionResult> Index()
         {
-            return View();
+            var adminUsers = await _userManager.GetUsersInRoleAsync("Admin");
+            var adminRole = _roleManager.Roles.Single(a => a.Name == "Admin").Id;
+            var userRole = _roleManager.Roles.Single(a => a.Name == "User").Id;
+
+            var users = _dbContext.Users.Include(a => a.Roles).ToList();
+            
+            var usersRoles = new List<UserRolesModel>();
+            foreach (var user in users)
+            {
+                var ur = new UserRolesModel();
+                ur.User = user;                
+                if (user.Roles.Any(a => a.RoleId == adminRole))
+                {
+                    ur.IsAdmin = true;
+                }
+                else
+                {
+                    ur.IsAdmin = false;
+                }
+                if (user.Roles.Any(a => a.RoleId == userRole))
+                {
+                    ur.IsUser = true;
+                }
+                else
+                {
+                    ur.IsUser = false;
+                }
+                usersRoles.Add(ur);
+            }           
+
+            return View(usersRoles);
+        }
+
+        public IActionResult ListNonAdminUsers()
+        {
+            var adminRole = _roleManager.Roles.Single(a => a.Name == "Admin").Id;
+            var users = _dbContext.Users.Include(a => a.Roles).Where(w => w.Roles.All(x => x.RoleId != adminRole)).ToList();
+
+            return View(users);
+        }
+
+        public IActionResult EditUser(string id)
+        {
+            var user = _dbContext.Users.SingleOrDefault(a => a.Id == id);
+            if (user == null)
+            {
+                ErrorMessage = "User Not Found";
+                return RedirectToAction("ListNonAdminUsers");
+            }
+
+            return View(user);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult EditUser(string id, [Bind("FirstName,LastName,Name,Phone,Account,ClientId")]User user)
+        {
+            var userToUpdate = _dbContext.Users.SingleOrDefault(a => a.Id == id);
+            if (userToUpdate == null)
+            {
+                ErrorMessage = "User Not Found";
+                return RedirectToAction("ListNonAdminUsers");
+            }
+            if (ModelState.IsValid)
+            {
+                userToUpdate.FirstName = user.FirstName;
+                userToUpdate.LastName = user.LastName;
+                userToUpdate.Name = user.Name;
+                userToUpdate.Phone = user.Phone;
+                userToUpdate.Account = user.Account;
+                userToUpdate.ClientId = user.ClientId;
```

Comment:
@jSylvestre If you are just going to copy over vars like this, you don't need to restrict the binding on the method signature.  If you want to restrict the binding like you are doing, then there is no need to copy since only the proper fields are being bound.  I'm fine with it this way but there are different options, and maybe we can get automapper in one day too ;)